src/d3a/models/strategy/update_frequency.py

```diff
@@ -30,25 +30,78 @@ def __init__(self, initial_rate, final_rate, fit_to_limit=True,
                  energy_rate_change_per_update=None, update_interval=duration(
                     minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL),
                  rate_limit_object=max):
-        self.fit_to_limit = fit_to_limit
-        self.initial_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                  initial_rate)
-        self.initial_rate = {}
-        self.final_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                final_rate)
-        self.final_rate = {}
+        self._fit_to_limit = fit_to_limit
+        self._initial_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                                   initial_rate)
+        self._initial_rate = {}
+        self._final_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                                 final_rate)
+        self._final_rate = {}
         if fit_to_limit is False:
-            self.energy_rate_change_per_update_profile_buffer = \
+            self._energy_rate_change_per_update_profile_buffer = \
                 read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_rate_change_per_update)
         else:
-            self.energy_rate_change_per_update_profile_buffer = {}
+            self._energy_rate_change_per_update_profile_buffer = {}
 
         self.energy_rate_change_per_update = {}
-        self.update_interval = update_interval
-        self.update_counter = {}
+        self._update_interval = update_interval
+        self._update_counter = {}
         self.number_of_available_updates = 0
         self.rate_limit_object = rate_limit_object
 
+    @property
+    def fit_to_limit(self):
+        return self._fit_to_limit
+
+    @fit_to_limit.setter
+    def fit_to_limit(self, fit_to_limit):
+        self._fit_to_limit = fit_to_limit
+
```

Comment:
Can you please explain me the rationale behind going from a self member to this property?
I see more code and an additional `self._fit_to_limit` and can not see the benefit of this. Thanks!

Reply:
This applies to all the other properties, too. A lot of boilerplate code.

Reply:
@hannesdiedrich this was requested by Hannes 

> Potentially scan through the Strategy classes to find occurrences where members of this class are directly accessed, and potentially add them to the class interface. That way we'll have more visibility over what the strategies require from these classes.
I'm not sure if he meant something else

Reply:
Just hopping in to ask: if you have setters, shouldn't they be used in these cases as well? https://github.com/gridsingularity/d3a/pull/1104/files/2d8a5e7b7f350cf7c421b0b957a4e6de32499f88#diff-bafdd7ead350b771695da62be3b1e612a379422c602b64f7f65a22746a2a78b6R33

Reply:
I added setters for the members that were set in the project ( For members that were only read, I only added getters )

Reply:
OK, I see. I am not sure whether this is what Spyros wanted, so leaving it to him to decide.

Reply:
This was only part of what I actually intended for this comment unfortunately :D To my eyes I do not have a problem with the extra boilerplate code for the setters (even though in this case they are no more useful than having a public member variable, since they contain no logic at all). The main purpose of my comment was to actually check where these are called from, and to see if we can group multiple property assignments in one method, in order to simplify the interface of the class. For instance, `initial_rate_profile_buffer` and `final_rate_profile_buffer` are both updated only in 2 places (in the constructor and in the area_reconfigure methods of the strategies), therefore we could expose a single method here that assigns both at once, instead of assigning them in the caller code.

Reply:
@spyrostz Should I export a generic update interface that can receive named parameters and update them ? because there are several combinations where we update different members

Reply:
@spyrostz

Reply:
Please have a look at the very end of each `_area_reconfigure_prices` method. You will see that this is the only place where the setters of these properties are invoked. This applies to all strategies (Load, PV, Storage).

Reply:
> @spyrostz Should I export a generic update interface that can receive named parameters and update them ? because there are several combinations where we update different members

Also if you can add a link with the part of the code where this actually happens it would be great, because from a brief search I did not find any such occurrences, but I might have missed it.

Reply:
@spyrostz Okay so we'll go with the one function updater and remove the property setters